### PR TITLE
add unit tests for Tab component

### LINKED ISSUE #111: (SP: 1) Write unit test for "Tab" component
### Component unit test
Unit test for "**Tab**" component
Scenaries descriptions:
- [ ] Label of the tab should be rendered
- [ ] Applies active tab if it is true
- [ ] Does not apply active tab if it is false
- [ ] Imitate click on the button and onClick function should be called

[Link to component](https://github.com/Space2Study-UA-1156/Client-01/blob/develop/src/components/tab/Tab.jsx)

Current coverage:
image

### PR DESCRIPTION
**Pull request by voloshynvv**
image

### CODE REVIEW

**Review by holus-bolus — APPROVED** [review]

**Review by OlhaOvsiienko — APPROVED** [review]

**Review by YaroslavChuiko — APPROVED** [review]

File: tests/unit/components/tab/Tab.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { screen, render, fireEvent } from '@testing-library/react'
+import { vi } from 'vitest'
+import Tab from '~/components/Tab/Tab'
+
+let label = 'label'
+let onClick = vi.fn()
```
Comment (vlad-khrychov) [inline tests/unit/components/tab/Tab.spec.jsx]:
```suggestion
const label = 'label'
const onClick = vi.fn()
```

File: src/components/tab/Tab.jsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { styles } from '~/components/tab/Tab.styles'
 const Tab = ({ activeTab, onClick, children }) => {
   return (
     <Button
+      aria-selected={activeTab}
```
Comment (vlad-khrychov) [inline src/components/tab/Tab.jsx]: Why it was added in PR with tests?

File: tests/unit/components/tab/Tab.spec.jsx
Comment (vlad-khrychov) [inline tests/unit/components/tab/Tab.spec.jsx:1]: Try to mock `Button from '@mui/material/Button'`

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

**Comment by vlad-khrychov** [comment]
Also, next time take screenshots with the coverage, please.

File: src/components/tab/Tab.jsx (hunk 2 above)
Comment (voloshynvv) [reply src/components/tab/Tab.jsx]: To control whether the Tab is active if the activeTab prop is true.
Should it be done in a different way? 

image

**Comment by voloshynvv** [comment]
image
image

File: src/components/tab/Tab.jsx (hunk 2 above)
Comment (vlad-khrychov) [reply src/components/tab/Tab.jsx]: So, if the button has been mocked and the `aria-selected` prop is not needed for the component `Tab`, we can remove this line. I think `activeTab && styles.activeTab` is enough. Or you can demonstrate it tomorrow.

File: tests/unit/components/tab/Tab.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { screen, render, fireEvent } from '@testing-library/react'
+import { vi } from 'vitest'
+import Tab from '~/components/Tab/Tab'
+
+vi.mock('~/components/tab/Tab.styles', () => ({
+  styles: {
+    defaultTab: {},
+    activeTab: 'active'
+  }
+}))
```
Comment (vlad-khrychov) [inline tests/unit/components/tab/Tab.spec.jsx:10]: It is good solution, I like it.

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**